thread-link.js

```diff
@@ -7,16 +7,33 @@
         var threadId = thread.dataset.topicId;
         var threadLink = document.URL + "/" + threadId;
         var follow = thread.querySelector('div[aria-label=Follow]');
-        if(!follow) { continue }
+        if (!follow) { continue }
         var threadLinkButtonId = 'permalink-' + threadId;
-        if(!document.getElementById(threadLinkButtonId)) {
+        if (!document.getElementById(threadLinkButtonId)) {
             var a = document.createElement('a');
             a.href = threadLink;
             a.title = "Permalink";
             a.innerText = "#";
             a.style = "padding: 0 0.3em 0 0.5em;font-weight: bold;";
             var followWrapper = follow.parentNode.parentNode;
             followWrapper.appendChild(a);
+
+            var span = document.createElement('span')
+            span.style = "cursor: pointer;"
+            span.title = "Copy permalink"
+            span.textContent = "@"
+            span.onclick = copyToClipboardFunc(threadLink);
+            followWrapper.appendChild(span);
         }
-    }    
-})();
\ No newline at end of file
+    }
+})();
+
+function copyToClipboardFunc(link) {
+    return function () {
+        navigator.clipboard.writeText(link).then(function () {
+            console.log("successfully copied text!")
```

Comment:
no need to log on success, I think we could just log on failures.

Reply:
Makes sense 👍